localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -159,6 +159,58 @@ def _get_current_unix_timestamp_utc():
 
             conn.commit()
 
+    def get_units_for_metric_data_stat(
+        self,
+        account_id: str,
+        region: str,
+        start_time: datetime,
+        end_time: datetime,
+        metric_name: str,
+        namespace: str,
+    ):
+        with sqlite3.connect(self.METRICS_DB) as conn:
+            cur = conn.cursor()
+
+            # prepare SQL query
+            start_time_unix = self._convert_timestamp_to_unix(start_time)
+            end_time_unix = self._convert_timestamp_to_unix(end_time)
+
+            data = (
+                account_id,
+                region,
+                namespace,
+                metric_name,
+                start_time_unix,
+                end_time_unix,
+            )
+
+            sql_query = f"""
+            SELECT GROUP_CONCAT(unit) AS unit_values
+            FROM(
+                SELECT
+                    DISTINCT COALESCE(unit, 'NULL_VALUE') AS unit
+                FROM (
+                    SELECT
+                    account_id, region, metric_name, namespace, timestamp, unit
+                    FROM {self.TABLE_SINGLE_METRICS}
+                    UNION ALL
+                    SELECT
+                    account_id, region, metric_name, namespace, timestamp, unit
+                    FROM {self.TABLE_AGGREGATED_METRICS}
+                ) AS combined
+                WHERE account_id = ? AND region = ?
+                AND namespace = ? AND metric_name = ?
+                AND timestamp >= ? AND timestamp < ?
+            ) AS subquery
+            """
+
+            cur.execute(
+                sql_query,
+                data,
+            )
+            result_row = cur.fetchone()
+            return result_row[0].split(",") if result_row[0] else ["NULL_VALUE"]
```

Comment:
could this theoretically result in an error if the result is not subscriptable?

sqlite3 docs:

> If [row_factory](https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.row_factory) is None, return the next row query result set as a [tuple](https://docs.python.org/3/library/stdtypes.html#tuple). Else, pass it to the row factory and return its result. Return None if no more data is available.

None[0] would result in an error I think

Reply:
@silv-io: thanks for the pointer :) i checked it once more:
The query will return `NULL` in case there are no data, meaning that the `cur.fetchone()` will return a tuple `(None,)`. The call should be safe therefore. 
Subsequent calls using `cur.fetchone()` would return `None` however (as there are no more rows to fetch).